src/controller/java/AndroidCallbacks-JNI.cpp

```diff
@@ -21,70 +21,135 @@
 #include <lib/support/CodeUtils.h>
 #include <lib/support/logging/CHIPLogging.h>
 
-#define JNI_METHOD(RETURN, CLASS_NAME, METHOD_NAME)                                                                                \
+#define JAVA_JNI_METHOD(RETURN, CLASS_NAME, METHOD_NAME)                                                                           \
     extern "C" JNIEXPORT RETURN JNICALL Java_chip_devicecontroller_##CLASS_NAME##_##METHOD_NAME
 
+#define KOTLIN_JNI_METHOD(RETURN, CLASS_NAME, METHOD_NAME)                                                                         \
+    extern "C" JNIEXPORT RETURN JNICALL Java_matter_controller_##CLASS_NAME##_##METHOD_NAME
+
 using namespace chip::Controller;
 
-JNI_METHOD(jlong, GetConnectedDeviceCallbackJni, newCallback)(JNIEnv * env, jobject self, jobject callback)
+JAVA_JNI_METHOD(jlong, GetConnectedDeviceCallbackJni, newCallback)(JNIEnv * env, jobject self, jobject callback)
+{
+    chip::DeviceLayer::StackLock lock;
+    GetConnectedDeviceCallback * connectedDeviceCallback = chip::Platform::New<GetConnectedDeviceCallback>(self, callback);
+    return reinterpret_cast<jlong>(connectedDeviceCallback);
+}
+
+JAVA_JNI_METHOD(void, GetConnectedDeviceCallbackJni, deleteCallback)(JNIEnv * env, jobject self, jlong callbackHandle)
+{
+    chip::DeviceLayer::StackLock lock;
```

Comment:
can both JAVA_JNI_METHOD and KOTLIN_JNI_METHOD share same implementation? I see implementation are same for same method across java and kotlin.

Reply:
I move seperate file related kotlin. And, implementation is moved to AndroidCallback.cpp file.